src/test/groovy/pl/sda/rentacar/domain/rent/RentServiceIntegrationSpec.groovy

```diff
@@ -0,0 +1,297 @@
+        given:
+        def departmentId = givenDepartmentExists(departmentRequest)
+        def employeeRequest = new EmployeeCreateRequest(
+                "Karol",
+                "Eklerek",
+                departmentId)
+        def employeeRequest2 = new EmployeeCreateRequest(
+                "Daniel",
+                "Riczardo",
+                departmentId)
+        employeeService.addEmployee(employeeRequest)
+        employeeService.addEmployee(employeeRequest2)
+        clientService.addClient(clientRequest)
+        clientService.addClient(clientRequest2)
+        carService.addCar(carRequest)
+        carService.addCar(carRequest2)
+        def request = new RentCreateRequest(
+                clientService.getALlClients().first().getId(),
+                employeeService.findAllEmployees().first().getId(),
+                carService.getALlCars().first().getId(),
+                "xD")
+        def request2 = new RentCreateRequest(
+                clientService.getALlClients().last().getId(),
+                employeeService.findAllEmployees().last().getId(),
+                carService.getALlCars().last().getId(),
+                "xDD")
+        service.addRent(request)
+        service.addRent(request2)
+
+        when:
+        def rents = service.getALlRents()
+
+        then:
+        rents !=null
+        rents.size() == 2
+        with(rents.last()) {
+            client.id == request2.clientId
+            employee.id == request2.employeeId
+            car.id == request2.carId
+            startDate == LocalDate.now()
+            returnDate == null
+            charge == null
+            comment == request2.comment
+            rentStatus == RentStatus.ACTIVE
+        }
+    }
+
+    def 'should update rent with given id'() {
+        given:
+        def rent = new RentCreateRequest(
+                clientService.getALlClients().first().getId(),
+                employeeService.findAllEmployees().first().getId(),
+                carService.getALlCars().first().getId(),
+                "this is comment")
+        def rentUpdate = new RentUpdateRequest(
+                LocalDate.now().plusDays(2L),
+                BigDecimal.valueOf(200/*carRequest2.getPricePerDay()*2*/),
+                "koniec imprezy",
+                RentStatus.FINISHED
+        )
+
+        service.addRent(rent)
+        def id = repository.findAll().last().id
+
+        when:
+        service.updateRent(id, rentUpdate)
+        def updatedRent = service.getRentById(id)
+
+        then:
+        with(updatedRent){
+            startDate == LocalDate.now()
+            returnDate == rentUpdate.returnDate
+            charge == rentUpdate.charge
+            comment == rentUpdate.comment
+            rentStatus == rentUpdate.rentStatus
+        }
+    }
+
+    def 'should remove rent with given id'(){
+        given:
+        def rent = new RentCreateRequest(
+                clientService.getALlClients().first().getId(),
```

Comment:
This test is using data saved to db by previous tests. This is incorrect. You should either initialize entire DB in the beginning of each test or use `setupSpec()` method https://spockframework.org/spock/docs/1.0/spock_primer.html
and do it once for all the tests